File: server/src/Models/DMToolkitModels/Definitions/BackgroundDefinition.cs
using DMToolkit.API.Models.DMToolkitModels.JoinTables;

namespace DMToolkit.API.Models.DMToolkitModels.Definitions;

public class BackgroundDefinition : IDefinition
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ICollection<AbilityScoreDefinition> AbilityScoreDefinitions { get; set; } = new List<AbilityScoreDefinition>();
    public string FeatDefinitionId { get; set; } = string.Empty;
    public FeatDefinition? FeatDefinition { get; set; } = null;
    public ICollection<SkillDefinition> SkillDefinitions { get; set; } = new List<SkillDefinition>();
    public ICollection<BackgroundDefinitionItemDefinitionBase> BackgroundDefinitionItemDefinitionBases { get; set; } = new List<BackgroundDefinitionItemDefinitionBase>();
    public int StartingGp { get; set; } = 50;
}

File: web-client/src/lib/types/dm-tool-types/definitions/backgroundDefinition.ts
import type { ItemDefinitionBaseQuantityDto } from "../relationships/itemDefinitionBaseQuantity";
import type { AbilityScoreDefinition } from "./abilityScoreDefinition";
import type { FeatDefinition } from "./featDefinition";
import type { SkillDefinition } from "./skillDefinition";

export interface BackgroundDefinition {
    id: string;
    name: string;
    description: string;
    abilityScoreDefinitions: AbilityScoreDefinition[];
    featDefinition: FeatDefinition | null;
    skillProficiencies: SkillDefinition[];
}

export interface BackgroundDefinitionDto {
    id: string;
    name: string;
    description: string;
    abilityScoreDefinitionIds: string[];
    featDefinitionId: string;
    skillDefinitionIds: string[]
    startingGp: number;
    itemDefinitionBaseQuantities: ItemDefinitionBaseQuantityDto[];
}